--- ../../build/gcc-10.2.0/libstdc++-v3/include/std/limits	2021-02-11 11:41:32.597596064 -0800
+++ /home/kzinti/opt/cross/x86_64-elf/include/c++/10.2.0/limits	2021-02-11 11:38:35.350193295 -0800
@@ -1655,6 +1655,8 @@

 #endif

+#if !(defined(__x86_64__) && _SOFT_FLOAT)
+
   /// numeric_limits<float> specialization.
   template<>
     struct numeric_limits<float>
@@ -1876,6 +1878,8 @@
 						      round_to_nearest;
     };

+#endif // #if !_SOFT_FLOAT
+
 #undef __glibcxx_long_double_has_denorm_loss
 #undef __glibcxx_long_double_traps
 #undef __glibcxx_long_double_tinyness_before
