--- ./local.h	2020-11-17 13:47:35.000000000 -0800
+++ ../../build/newlib-4.0.0/newlib/libc/stdio/local.h	2021-01-19 22:02:34.182169807 -0800
@@ -56,7 +56,8 @@
    the appropriate _newlib_XXX_exit macro. */

 #if !defined (__SINGLE_THREAD__) && defined (_POSIX_THREADS) \
-    && !defined (__rtems__)
+    && !defined (__rtems__) \
+    && !defined (__rainbow__) // TODO: we do want _STDIO_WITH_THREAD_CANCELLATION_SUPPORT defined for user space!!!
 #define _STDIO_WITH_THREAD_CANCELLATION_SUPPORT
 #endif

