--- config.gcc	2021-03-27 14:36:24.372456927 -0700
+++ ../../build/src/gcc-10.2.0/gcc/config.gcc	2021-03-27 14:51:04.324818362 -0700
@@ -1038,6 +1038,10 @@
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-errata.h aarch64/aarch64-elf-raw.h"
 	tmake_file="${tmake_file} aarch64/t-aarch64"
 	case $target in
+	aarch64-rainbow-elf*)
+		use_gcc_stdint=wrap
+		tm_file="${tm_file} rainbow.h"
+		;;
 	aarch64-*-elf*)
 		use_gcc_stdint=wrap
 		;;
@@ -1308,6 +1312,11 @@
 	tmake_file="${tmake_file} arm/t-arm arm/t-arm-elf"
 	target_cpu_cname="arm7tdmi"
 	case ${target} in
+	arm*-rainbow-eabi*)
+	  tm_file="${tm_file} newlib-stdint.h rainbow.h"
+	  tmake_file="${tmake_file} arm/t-bpabi"
+	  use_gcc_stdint=wrap
+	  ;;
 	arm*-*-eabi*)
 	  tm_file="$tm_file newlib-stdint.h"
 	  tmake_file="${tmake_file} arm/t-bpabi"
@@ -1861,9 +1870,16 @@
 i[34567]86-*-elfiamcu)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/iamcu.h"
 	;;
+i[34567]86-rainbow-elf*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h rainbow.h"
+	;;
 i[34567]86-*-elf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h"
 	;;
+x86_64-rainbow-elf*)
+	tmake_file="${tmake_file} i386/t-x86_64-rainbow"
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h rainbow.h"
+	;;
 x86_64-*-elf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h"
 	;;
